yamale/schema/schema.py

```diff
@@ -55,14 +56,7 @@ def _parse_schema_item(self, path, expression, validators):
     def validate(self, data, data_name, strict):
         path = DataPath()
         errors = self._validate(self._schema, data, path, strict)
-
```

Comment:
It seems there is either a validation error or there isn't. Can you explain why we need a result object rather than just raising an exception here?

Reply:
Answer bellow